source/slang/slang-capabilities.capdef

```diff
@@ -397,3 +448,83 @@ alias DX_6_4 = sm_6_4;
 alias DX_6_5 = sm_6_5;
 alias DX_6_6 = sm_6_6;
 alias DX_6_7 = sm_6_7;
+
+alias sm_2_0_GLSL_140 = sm_4_0 | glsl | spirv_1_0 | cuda | cpp;
+alias sm_2_0_GLSL_400 = sm_4_0 | glsl | spirv_1_0 | cuda | cpp;
+alias appendstructuredbuffer = sm_5_0 + shader_stages_fra_com;
+alias atomic_hlsl = _sm_4_0 | GLSL_430_SPIRV_1_0;
+alias atomic_hlsl_nvapi = _sm_4_0 + hlsl_nvapi;
+alias atomic_hlsl_sm_6_6 = _sm_6_6;
+alias byteaddressbuffer = sm_4_0;
+alias byteaddressbuffer_rw = sm_4_0 + shader_stages_fra_com;
+alias consumestructuredbuffer = sm_5_0 + shader_stages_fra_com;
+alias fragmentprocessing = fragment + sm_5_0;
+alias fragmentprocessing_derivativecontrol = fragment + sm_5_0 + GL_ARB_derivative_control;
+alias getattributeatvertex = fragment + _sm_6_1 | fragment + GL_EXT_fragment_shader_barycentric;
+alias memorybarrier_com = compute + sm_5_0;
+alias structuredbuffer = sm_4_0;
+alias structuredbuffer_rw = sm_4_0 + shader_stages_fra_com;
+alias texture_sm_4_1 = sm_4_1;
+alias texture_sm_4_1_fra = sm_4_1 + fragment;
+alias texture_sm_4_1_clamp_fra = texture_sm_4_1_fra;
+alias texture_sm_4_1_ver_fra_geo = sm_4_1 + shader_stages_ver_fra_geo;
+
+alias atomic_glsl_float1 = GL_EXT_shader_atomic_float;
+alias atomic_glsl_float2 = GL_EXT_shader_atomic_float2;
+alias atomic_glsl_halfvec = GL_NV_shader_atomic_fp16_vector;
+alias atomic_glsl = GLSL_430_SPIRV_1_0;
+alias atomic_glsl_int64 = atomic_glsl + GL_EXT_shader_atomic_int64;
+alias GLSL_430_SPIRV_1_0_com = GLSL_420_SPIRV_1_0 + compute;
+alias image_samples = GLSL_450 | spirv_1_0 + spvImageQuery | sm_4_1;
+alias image_query = GLSL_430 | GLSL_430 + spvImageQuery;
+alias image_loadstore = GL_EXT_shader_image_load_store + GLSL_420;
+alias nonuniformqualifier = sm_5_1;
+alias printf = GL_EXT_debug_printf | _sm_4_0 | _cuda_sm_2_0 | cpp;
+alias texturefootprint = GL_NV_shader_texture_footprint + GLSL_450 | hlsl_nvapi + _sm_4_0;
+alias texturefootprintclamp = texturefootprint + GL_ARB_sparse_texture_clamp;
+alias texture_shadowlod = GL_EXT_texture_shadow_lod + GLSL_140;
+alias texture_cube = GLSL_400;
+alias shader5_sm_4_0 = GL_ARB_gpu_shader5 | _sm_4_0 | _cuda_sm_2_0 | cpp;
```

Comment:
Probably this is correct but I am wondering if this uses GL_ARB_gpu_shader5 not 4?

Reply:
glsl 130 implements shader4 (mostly just adds Uint and Int support).

Slang only supports Vulkan officially for now. [As per Vulkan GLSL spec](https://github.com/KhronosGroup/GLSL/blob/main/extensions/khr/GL_KHR_vulkan_glsl.txt) anything at or below GLSL 140 is assumed to be supported if using Vulkan.

For old OpenGL support (if Slang is to support GLSL 130), it would mostly rely on the compiler enabling shader4 if Uint/Int types are found for some special annotated functions.